src/test/java/org/minions/devfund/lourdes/WhackaMoleTest.java

```diff
@@ -0,0 +1,140 @@
+package org.minions.devfund.lourdes;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test for Test for {@link WhackAMole}.
+ */
+public class WhackaMoleTest {
+
+    private WhackAMole whackAMole;
+
+    /**
+     * Method that setup initial values.
+     */
+    @Before
+    public void setup() {
+        final int numAttempts = 50;
+        final int gridDimension = 10;
+        whackAMole = new WhackAMole(numAttempts, gridDimension);
+    }
+
+    /**
+     * Verify that a Mole is placed in the given position.
+     */
+    @Test
+    public void testPlaceMole() {
+        final int positionX = 6;
+        final int positionY = 5;
+        assertTrue(whackAMole.place(positionX, positionY));
+    }
+
+    /**
+     * Verify that a Mole is not assigned in a position that has been already assigned.
+     */
+    @Test
+    public void testPlaceMoleInSamePosition() {
+        final int positionX = 1;
+        final int positionY = 2;
+        whackAMole.place(positionX, positionY);
+        assertFalse(whackAMole.place(positionX, positionY));
+    }
+
+    /**
+     * Verify a whack is done in the given position.
+     */
+    @Test
+    public void testWhack() {
```

Comment:
Since your pull request is still open.
please write your unit tests something like this.

Running: testConstructorGrid
Finished: testConstructorGrid
Running: testConstructorNumOfAttempts
Finished: testConstructorNumOfAttempts
Running: testPlaceFail
Finished: testPlaceFail
Running: testWhackSucceedAttempts
Finished: testWhackSucceedAttempts
Running: testWhackFailAttempts
Finished: testWhackFailAttempts
Running: testPlaceSucceed
Finished: testPlaceSucceed
Running: testWhackFailMolesLeft
Finished: testWhackFailMolesLeft
Running: testWhackSucceedMolesLeft
Finished: testWhackSucceedMolesLeft